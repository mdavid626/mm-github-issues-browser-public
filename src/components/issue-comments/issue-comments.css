.IssueComments {
  display: flex;
  flex-direction: column;
  gap: var(--space3);
}

.IssueComments-title {
  font-size: 20px;
  font-weight: bold;
}

.IssueComments-comments {
  display: flex;
  flex-direction: column;
  gap: var(--space3);
}

.IssueComments-loadMore {
  cursor: pointer;
  font-size: 18px;
  font-weight: bold;
  align-self: center;
  color: var(--primary-color);
}

.IssueComments-loading {
  font-size: 18px;
  font-weight: bold;
  align-self: center;
  color: var(--primary-color);
}