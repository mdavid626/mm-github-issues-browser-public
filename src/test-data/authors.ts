import { Author } from '../types/author';

export const testAuthor1: Author = {
  login: 'testAuthor1',
};
