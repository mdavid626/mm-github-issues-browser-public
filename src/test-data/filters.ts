import { Filters } from '../types/filters';

export const testFilters1: Filters = {
  search: 'test',
  state: 'open',
};
