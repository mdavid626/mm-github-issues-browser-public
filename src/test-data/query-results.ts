export const issueQueryResult = {
  data: {
    repository: {
      id: 'MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==',
      issue: {
        id: 'I_kwDOAJy2Ks5ePiI3',
        createdAt: '2023-02-12T05:31:50Z',
        title: 'Bug: build args parse error on powsershell',
        state: 'CLOSED',
        number: 26156,
        stateReason: 'COMPLETED',
        url: 'https://github.com/facebook/react/issues/26156',
        body: 'test issue result',
        author: { login: 'weidehai', __typename: 'User' },
        comments: {
          nodes: [],
          pageInfo: {
            hasNextPage: false,
            endCursor: null,
            __typename: 'PageInfo',
          },
          totalCount: 0,
          __typename: 'IssueCommentConnection',
        },
        __typename: 'Issue',
      },
      __typename: 'Repository',
    },
  },
};
