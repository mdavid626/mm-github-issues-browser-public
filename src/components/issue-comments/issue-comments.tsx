import React from 'react';
import { useFetchMoreIssueComment } from '../../hooks/issue-hooks/issue-hooks';
import { Issue } from '../../types/issue';
import IssueCommentsItem from '../issue-comments-item/issue-comments-item';
import './issue-comments.css';

const IssueComments: React.FC<{
  issueNumber: number;
  comments: Issue['comments'];
}> = ({ issueNumber, comments }) => {
  const [fetchMoreComment, isFetchingMoreComment] =
    useFetchMoreIssueComment(issueNumber);
  return (
    <div className="IssueComments">
      <div className="IssueComments-title">
        Comments ({comments.totalCount})
      </div>
      {comments.nodes.length === 0 ? (
        <div>No comments yet</div>
      ) : (
        <div className="IssueComments-comments">
          {comments.nodes.map((comment) => (
            <IssueCommentsItem key={comment.id} comment={comment} />
          ))}
        </div>
      )}
      {comments.pageInfo.hasNextPage && !isFetchingMoreComment && (
        <div
          onClick={() => fetchMoreComment(comments.pageInfo.endCursor)}
          className="IssueComments-loadMore"
        >
          load more comments
        </div>
      )}
      {isFetchingMoreComment && (
        <div className="IssueComments-loading">loading...</div>
      )}
    </div>
  );
};

export default IssueComments;
