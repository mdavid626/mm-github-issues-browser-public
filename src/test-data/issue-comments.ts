import { IssueComment, IssueComments } from '../types/issue-comment';
import { testAuthor1 } from './authors';

export const testIssueComment1: IssueComment = {
  id: 'IC_kwDOAJy2Ks5Ux_nZ',
  body: 'issue comment 1',
  createdAt: '2023-02-12T09:43:24.641Z',
  author: testAuthor1,
};

export const testIssueComment2: IssueComment = {
  id: 'IC_kwDOAJy2Ks5Ux4u7',
  body: 'issue comment 2',
  createdAt: '2022-01-11T10:42:25.655Z',
  author: testAuthor1,
};

export const testIssueComments1: IssueComments = {
  nodes: [testIssueComment1, testIssueComment2],
  pageInfo: {
    hasNextPage: true,
    endCursor: 'endCursor1',
  },
  totalCount: 2,
};

export const testEmptyIssueComments: IssueComments = {
  nodes: [],
  pageInfo: {
    hasNextPage: false,
    endCursor: 'endCursor1',
  },
  totalCount: 0,
};
