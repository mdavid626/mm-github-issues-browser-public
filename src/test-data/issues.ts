import { Issue, IssueQueryResult, IssuesItem } from '../types/issue';
import { testAuthor1 } from './authors';
import { testIssueComments1 } from './issue-comments';

export const testIssue1: Issue = {
  id: 'I_kwDOAJy2Ks5d65oG',
  createdAt: '2023-02-08T08:50:14Z',
  title: 'Bug: Failed when set homepage in package.json or set PUBLIC_URL',
  state: 'CLOSED',
  number: 26126,
  url: 'https://github.com/facebook/react/issues/26126',
  body: 'test issue1 body',
  author: testAuthor1,
  comments: testIssueComments1,
};

export const testIssuesItem1: IssuesItem = {
  id: 'I_kwDOAJy2Ks5d65oG',
  createdAt: '2023-02-08T08:50:14Z',
  title: 'Bug: Failed when set homepage in package.json or set PUBLIC_URL',
  state: 'CLOSED',
  number: 26126,
  url: 'https://github.com/facebook/react/issues/26126',
  author: testAuthor1,
};

export const testIssueQueryResult1: IssueQueryResult = {
  repository: {
    issue: testIssue1,
  },
};
