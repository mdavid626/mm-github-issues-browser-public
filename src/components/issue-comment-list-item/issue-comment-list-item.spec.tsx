import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';
import { testIssueComment1 } from '../../test-data/issue-comments';
import IssueCommentListItem from './issue-comment-list-item';

describe('issue-comment-list-item', () => {
  afterEach(cleanup);

  it('should render', () => {
    const { asFragment } = render(
      <IssueCommentListItem comment={testIssueComment1} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render custom class name', () => {
    render(
      <IssueCommentListItem
        comment={testIssueComment1}
        className="myClassName"
      />
    );
    expect(screen.getByTestId('IssueCommentListItem')).toHaveClass(
      'myClassName'
    );
  });
});
