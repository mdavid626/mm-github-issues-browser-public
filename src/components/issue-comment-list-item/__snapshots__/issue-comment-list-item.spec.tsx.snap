// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-comment-list-item should render 1`] = `
<DocumentFragment>
  <div
    class="IssueCommentListItem"
    data-testid="IssueCommentListItem"
  >
    <div
      class="IssueCommentListItem-detail"
    >
      <div>
        testAuthor1
      </div>
      <div>
        12.02.2023 10:43
      </div>
    </div>
    <div
      class="IssueCommentListItem-comment"
    >
      issue comment 1
    </div>
  </div>
</DocumentFragment>
`;
